Sources/App/Core/Extensions/SiteMap+ResponseEncodable.swift

```diff
@@ -15,11 +15,18 @@
 import Plot
 import Vapor
 
+extension SiteMapIndex: AsyncResponseEncodable {
+    public func encodeResponse(for request: Request) -> Response {
+        let res = Response(status: .ok, body: .init(string: self.render()))
+        res.headers.add(name: "Content-Type", value: "text/xml")
+        return res
+    }
+}
 
-extension SiteMap: ResponseEncodable {
-    public func encodeResponse(for request: Request) -> EventLoopFuture<Response> {
+extension SiteMap: AsyncResponseEncodable {
+    public func encodeResponse(for request: Request) -> Response {
         let res = Response(status: .ok, body: .init(string: self.render()))
         res.headers.add(name: "Content-Type", value: "text/xml")
-        return request.eventLoop.makeSucceededFuture(res)
+        return res
     }
 }
```

Comment:
Perhaps:

```swift
extension SiteMapIndex: AsyncResponseEncodable, Renderable {
    public func encodeResponse(for request: Request) async throws -> Response {
        try await encodeXMLResponse(for: request)
    }
}

extension SiteMap: AsyncResponseEncodable, Renderable {
    public func encodeResponse(for request: Request) async throws -> Response {
        try await encodeXMLResponse(for: request)
    }
}

extension AsyncResponseEncodable where Self: Renderable {
    func encodeXMLResponse(for request: Request) async throws -> Response {
        let res = Response(status: .ok, body: .init(string: self.render()))
        res.headers.add(name: "Content-Type", value: "text/xml")
        return res
    }
}
```

Also note that `AsyncResponseEncodable` protocol req for `encodeResponse` is really `async throws`. That's why this change will trigger a `try await` missing error in `SiteMapController.buildStaticPages`.

Reply:
Fixed.